music21/musicxml/xmlToM21.py

```diff
@@ -867,6 +867,21 @@ def xmlRootToScore(self, mxScore, inputM21=None):
             self.spannerBundle.remove(sp)
 
         s.coreElementsChanged()
+        # Fill gaps with rests where needed
+        for m in s[stream.Measure]:
+            for v in m.voices:
+                if v:  # do not bother with empty voices
+                    # the musicDataMethods use insertCore, thus the voices need to run
+                    # coreElementsChanged
+                    v.coreElementsChanged()
```

Comment:
Note to self: if we decide to leave this here after all, this is probably not needed (but still think MeasureParser is better if I'm not missing something)

Reply:
It's been a while since I've thought through this PR, but I think that this is correct.